telethon/events/raw.py

```diff
@@ -46,6 +46,8 @@ def build(cls, update, others=None, self_id=None):
         return update
 
     def filter(self, event):
-        if ((not self.types or isinstance(event, self.types))
-                and (not self.func or self.func(event))):
+        if not self.types or isinstance(event, self.types):
+            if self.func:
+                # Return the result of func directly as it may need to be awaited
+                return self.func(event)
             return event
```

Comment:
We're not returning true or false, but the event. Anyway, it's fine as long as it works and the docs now say truthy, not `True`. Just pointing it out.

Reply:
Ya, forgot to change, you are right. Still - it doesn't change anything ;D